source/wifi/wifi_hal.c

```diff
@@ -9524,7 +9524,7 @@ INT wifi_getApAssociatedDevice(INT ap_index, CHAR *output_buf, INT output_buf_si
     if (!status)
         return RETURN_OK;
 
-    sprintf(cmd, "hostapd_cli -i %s%d list_sta | tr -d \"\\n\"", AP_PREFIX, ap_index);
+    sprintf(cmd, "hostapd_cli -i %s%d list_sta | tr '\\n' ',' | sed 's/.$//'", AP_PREFIX, ap_index);
```

Comment:
Hello @BroLeaf ,
Can you please share the content of output buffer. because i am getting different results with this logic.

i slightly modified your logic like this.
`| tr '$\n' ',' | sed 's/.$//'`

it is working for me.

Thanks.

Reply:
Hi @manigandanj ,
I got the same output by my method and the method you given.
I connect two device, the result is `14:xx:xx:xx:xx:04,ce:xx:xx:xx:xx:f4` without any newline character.

Do you directly execute the command? Maybe you forget to remove the escape character in the C code.
If the result still not the same, I will modify the implementation.

Thanks.

Reply:
Hello @BroLeaf ,
If you got correct value in the output buffer, i am very much okay.

Thank you so much for your contribution. Approving it.